CONTRIBUTING.md

```diff
@@ -85,6 +85,25 @@ All PRs require an approval from at least one CODEOWNER before merge. PRs which
 - If you sat down with the PR submitter and did a pairing review please note that in the `Approval`, or your PR comments.
 - If you are only making "surface level" reviews, submit any notes as `Comments` without adding a review.
 
+### Commit Messages
+
+Commit messages should be [conventional](https://www.conventionalcommits.org/en/v1.0.0/).
+
+The commit message type should be one of:
+
+* `feat` / `feature` for feature work.
+* `bug` / `fix` for bug fixes.
+* `imp` / `improvements` for improvements.
+* `doc` / `docs` / `documentation` for any documentation changes.
+* `test` / `e2e` for addition or improvements of unit, integration and e2e tests or their corresponding infrastructure.
+* `deprecated` for deprecation changes.
+* `deps` / `build` for changes to dependencies.
+* `chore` / `misc` / `nit` for any miscellaneous changes that don't fit into another category.
+
+**Note**: If any change breaking, the following format must be used:
+* `type` + `(api)!` for api breaking changes, e.g. `fix(api)!: api breaking fix`
+* `type` + `(statemachine)!` for state machine breaking changes, e.g. `fix(statemachine)!: state machine breaking fix`
```

Comment:
we could also use `consensus`. consensus breaking might be more intuitive? I have no preference

Reply:
I have no preference, from personal experience I hear and see the term `statemachine` breaking far more often, I'm not sure if this is true for external contributors. WDYT?

Reply:
sounds good, I think it is what we have in our release document as well